### ISSUE TITLE
Active state for headings not working when there are additional attributes

### ISSUE THREAD

**Issue by timc1** [remirror/remirror#385]
## Description

This might be what caused my issue with #379 🤔

I have an extension which appends a unique id to top-level blocks, something like:

```
<h1 id="unique-id-1>Hello!</h1>
<p id="unique-id-2>Some text here</p>
<ul id="unique-id-3">
  <li><p>Item</p></li>
  <li><p>Item</p></li>
</ul>
```

At this point, when triggering the menu for `<h1 id="unique-id-1>Hello!</h1>`, the active state for H1 is false.

It looks like [`isNodeActive`](https://github.com/remirror/remirror/blob/9dafc29cc7a70ec1c9e5cbd19091f27e20d7b035/packages/%40remirror/core-utils/src/prosemirror-utils.ts#L535-L549) returns `false` when there are extra `attrs` other than exactly `{ level: <number> }`; in my case, `{ id: <string>, level: <number> }`

**Returns `false` when there is an additional attribute, `id`:**

<img width="379" alt="Screen Shot 2020-07-27 at 3 03 05 PM" src="https://user-images.githubusercontent.com/12195101/88580946-6aef0d80-d01a-11ea-9d2b-6b15b00a6b37.png">

**Returns `true` as expected when `level` is the only attribute:**

<img width="379" alt="Screen Shot 2020-07-27 at 3 04 46 PM" src="https://user-images.githubusercontent.com/12195101/88581020-8d812680-d01a-11ea-96a5-2f0b6dde2706.png">

## Checklist

- [x] I have read the [**contributing**](https://github.com/remirror/remirror/blob/HEAD/docs/contributing.md) document.

**Comment by tommoor** [comment]
This is default behavior or prosemirror's `hasMarkup` method, to do a deep comparison: 
https://github.com/ProseMirror/prosemirror-model/blob/0a1decebf98c4c3cb4926b848d569ce4b7dfa2ff/src/node.js#L131-L135

Definitely unworkable for this usecase of unique ID's though 🤔

**Comment by ifiokjr** [comment]
Thanks for opening this. Based on your comment @tommoor, I've come up with a possible fix outlined in #386
